airbyte-ci/connectors/pipelines/pipelines/dagger/actions/gradle.py

```diff
@@ -0,0 +1,230 @@
+    GRADLE_DEP_CACHE_PATH = "/root/gradle-cache"
+    GRADLE_HOME_PATH = "/root/.gradle"
+    LOCAL_MAVEN_REPOSITORY_PATH = "/root/.m2"
+    YUM_PACKAGES_TO_INSTALL = [
+        "docker",  # required by :integrationTestJava.
+        "findutils",  # gradle requires xargs, which is shipped in findutils.
+        "jq",  # required by :acceptance-test-harness to inspect docker images.
+        "rsync",  # required for gradle cache synchronization.
+    ]
+
+    def __init__(
+        self,
+        dagger_client: dagger.Client,
+        gradle_project_dir: dagger.Directory,
+        local_execution: bool,
+        build_cdk: bool = False,
+        export_gradle_project_directory_to_host: bool = False,
+        export_gradle_project_directory_to_host_path: str = ".",
+        workdir: str = "/airbyte",
+        s3_build_cache_access_key_id: Optional[dagger.Secret] = None,
+        s3_build_cache_secret_key: Optional[dagger.Secret] = None,
+    ):
+        """Initialize a gradle task executor.
+
+        Args:
+            dagger_client (dagger.Client): The dagger client.
+            gradle_project_dir (dagger.Directory): The gradle project directory to run gradle in.
+            local_execution (bool): Whether the gradle task executor is running locally.
+            build_cdk (bool): Whether to build and publish the local CDK if needed. Defaults to False.
+            export_gradle_project_directory_to_host (bool, optional): Whether to export the gradle project directory to the host on context exit. Defaults to False.
+            export_gradle_project_directory_to_host_path (str, optional): The path to where the gradle project directory will be exported on the host filesystem. Defaults to ".".
+            workdir (str, optional): The container path where gradle_project_dir will be mounted. Defaults to "/airbyte".
+            s3_build_cache_access_key_id (Optional[dagger.Secret], optional): The S3 build cache access key ID as a dagger secret. Defaults to None.
+            s3_build_cache_secret_key (Optional[dagger.Secret], optional): The S3 build cache secret key as a dagger secret. Defaults to None.
+        """
+        self.dagger_client = dagger_client
+        self.gradle_project_dir = gradle_project_dir
+        self.local_execution = local_execution
+        self.build_cdk = build_cdk
+        self.export_gradle_project_directory_to_host = export_gradle_project_directory_to_host
+        self.export_gradle_project_directory_to_host_path = export_gradle_project_directory_to_host_path
+        self.workdir = workdir
+        self.s3_build_cache_access_key_id = s3_build_cache_access_key_id
+        self.s3_build_cache_secret_key = s3_build_cache_secret_key
+
+    @property
+    def dependency_cache_volume(self) -> dagger.CacheVolume:
+        """This cache volume is for sharing gradle dependencies (jars and poms) across all pipeline runs."""
+        return self.dagger_client.cache_volume(self.DEPENDENCY_CACHE_VOLUME_NAME)
```

Comment:
Just curious why do we want to have this cache volume as a property when we're never going to want to reference it outside the class.